samples/saga/batching/sample.md

```diff
@@ -0,0 +1,61 @@
+---
+title: Batch messages using Saga
+summary: Use batching when generating tasks using Saga
+reviewed: 2022-04-09
+component: Core
+related:
+- nservicebus/sagas
+---
+
+
+## Introduction
+
+This sample shows how to control the task (work) generation via batching of the work items through Sagas.
+
+Once the projects are started, press <kbr>S</kbr> to send a `StartProcessing` message with a randomly `WorkCount` number that starts a new saga instance. This is a result of as configuration specified with the `IAmStartedByMessages` interface and mapping that [correlates](/nservicebus/sagas/message-correlation.md) `StartProcessing.ProcessId` property on the message with `OrderSagaData.ProcessId` property on the saga data.
+
+When processing the `StartProcessing` message each saga instance creates `ProcessWorkOrder` in batches of 100 messages (in each batch) and sends them off to the `WorkProcessor` endpoint to handle. Once all the messages of a batch is completed, the next batch of messages are generated.
+
+The output to the console of the `WorkGenerator` will be:
+
+```
+Started.
+Press 'S' to start a new process or [Enter] to exit.
+SStarted process 'cc39af80-d33c-45e0-9991-5ece40cf7d6b' with '473' work orders.
+Processing saga started: 'cc39af80-d33c-45e0-9991-5ece40cf7d6b'
+Starting the process for '473' work orders.
+Queueing next batch of work orders: (1 - 100).
+Queueing next batch of work orders: (101 - 200).
+Queueing next batch of work orders: (201 - 300).
+Queueing next batch of work orders: (301 - 400).
+Queueing next batch of work orders: (401 - 473).
+All done. Took 33.737792
+```
+
+The output to the console of the `WorkProcessor` will be:
+
+```
+Started.
+Press [Enter] to exit.
+Processing work order '1'
+Processing work order '4'
+Processing work order '10'
+Processing work order '2'
+...
+Processing work order '431'
+Processing work order '427'
+Processing work order '441'
+Processing work order '420'
+```
+
+Note that while the work requests are generated in sequence, on the processing side the work can happen out of order. 
```

Comment:
One of the drawbacks that I think need to be documented is. What when one or more item fails processing? That would result in the 'paged processing' to fail.

I think it would be good to keep in mind that 'paged processing' should not fail due to non-transient errors and that the invoked handler ensures a "Reply"

Reply:
I think this would cover that:

>The saga controls the generation of the next batch of work and monitors for the completion signal of all messages in that batch, before starting the next batch.

it wouldn't necessarily fail the paged processing but would stall it and won't move to the subsequent page. If that is not desired, the worker process can send a different kind of 'reply' that contains error info. In other words, this is a basic structure without being too prescriptive as that'd be different based on business requirements.

Reply:
I understand, but I do think its important to highlight issues that could cause harm if not dealt with during the design. I added a suggestion.